core/trino-parser/src/main/java/io/trino/sql/tree/SetTimeZone.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.tree;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public class SetTimeZone
+        extends Statement
+{
+    private final Optional<Expression> timeZone;
+
+    public SetTimeZone()
+    {
+        this(Optional.empty(), Optional.empty());
+    }
+
+    public SetTimeZone(Expression timeZone)
+    {
+        this(Optional.empty(), Optional.of(requireNonNull(timeZone, "timeZone is null")));
```

Comment:
Move the `requireNonNull` checks to the private constructor.

Reply:
I think that, even though I have the duplication in the two constructors, using  `requireNonNull(timeZone, "timeZone is null")` provides a much better exception handling.
Doing the change you suggest, I'd have to work with `Optional.ofNullable(...)` and in case of providing a `NULL` expression I'd default to `LOCAL` `SetTimeZone` silently swallowing a potentional NPE.

Reply:
Alternatively, you could have a public constructor taking an `Optional<Expression>` instead of separate constructors with or without the expression. Then the private constructor would perform the `requireNonNull` check on that argument.